src/modules/departments/ui/departments-dialog.tsx

```diff
@@ -0,0 +1,47 @@
+import { useTranslation } from "react-i18next";
+import { Button } from "antd";
+import { Form } from "@/components/form";
+import { FormTextField } from "@/components/form-text-field";
+import { createDialogHook } from "@/helpers/dialog/create-dialog";
+import BaseDialog from "@/components/dialog/base-dialog";
+import { getDepartmentFormSchema } from "../shemas/department";
+import { FormSubmitButton } from "@/components/form-submit-button";
+
+interface FormData {
+    department: string;
+}
+interface DepartmentDialogProps {
+    title: string;
+    onConfirm: (formData: FormData) => void;
+    onClose: () => void;
+    initialValues: { department: string };
+}
+const DepartmentDialog = ({ title, onConfirm, onClose, initialValues }: DepartmentDialogProps) => {
+    const { t } = useTranslation();
+    const { department } = initialValues;
+
+    const handleConfirm = (formData: FormData) => {
+        onConfirm(formData);
+        onClose();
+    };
+
+    return (
+        <BaseDialog title={title} onClose={onClose}>
+            <Form
+                onSubmit={handleConfirm}
+                initialValues={{ department }}
```

Comment:
initialValues is just an AntdForm prop, so it may not be properly handled by react-hook-form here. Probably should replace it with defaultValues (which is actually utilised)

And also the submit button propably doesn't need to be enabled if the default (initial) values haven't changed (and submiting a form doesn't make much sense), so disableIfNotDirty may also be applied

Reply:
@RC-3222 
can you explain in which case we can have troubles  between AntdForm prop and react-hook-form. I did not understand why we shoud use custom prop defaultValues.

Thanks for advice with disabling submit button

Reply:
First, defaultValues' type is synchronized with the schema passed to the form (so there's some typchecking for the fields), second - as it is basically a defaultValues option from useForm here, the defult values (passed as initialValues) may not be handled properly by form resetting
(and even though these props could be synchronized, just why?)

Reply:
ok, done